### PR TITLE
PS-8978: Use different `WORKER_x_MTR_SUITES` for `Debug` and `RelWithDebInfo`

### PR DESCRIPTION
**Pull request by inikep**
1. Set different `WORKER_x_MTR_SUITES` for different `CMAKE_BUILD_TYPE`.
2. Improve logging for `setupTestSuitesSplit()`.

### CODE REVIEW

File: jenkins/suites-groups.sh [hunk 1]
```diff
@@ -132,11 +149,5 @@ check() {
   fi
   echo "************************"
 
-  exit ${failure}
+  return ${failure}
 }
-
-case "$1" in
```
Comment (kamil-holubicki) [inline jenkins/suites-groups.sh:138]: I would leave this part. It allows us to run the script locally just to check if everything is OK without the need of sourcing it to the current shell.

File: jenkins/pipeline-parallel-mtr.groovy [hunk 2]
```diff
@@ -238,7 +238,9 @@ void setupTestSuitesSplit() {
             wget \${RAW_VERSION_LINK}/${BRANCH}/mysql-test/mysql-test-run.pl -O ${WORKSPACE}/mysql-test-run.pl
             chmod +x ${WORKSPACE}/suites-groups.sh
             set +e
-            ${WORKSPACE}/suites-groups.sh check ${WORKSPACE}/mysql-test-run.pl
+            # Set WORKER_x_MTR_SUITES and check if suites list is consistent with the one specified in mysql-test-run.pl
+            source ${WORKSPACE}/suites-groups.sh
```
Comment (kamil-holubicki) [inline jenkins/pipeline-parallel-mtr.groovy]: Yes, we source it to get split variables, but I'm not sure if it is the right way to call the function directly from sourced script. In my opinion more readable is calling a script as it was done before

File: jenkins/pipeline-parallel-mtr.groovy [hunk 3]
```diff
@@ -238,7 +238,9 @@ void setupTestSuitesSplit() {
...
+            check_suites ${CMAKE_BUILD_TYPE} ${WORKSPACE}/mysql-test-run.pl
```
Comment (kamil-holubicki) [inline jenkins/pipeline-parallel-mtr.groovy]: Check should not depend on build type, right? We always use the same set of suites.

File: jenkins/suites-groups.sh (hunk 1 above)
Comment (inikep) [reply jenkins/suites-groups.sh:138]: I've never seen such type of a call:
```
${WORKSPACE}/suites-groups.sh check ${WORKSPACE}/mysql-test-run.pl
```
The traditional usage is to `source` functions from other scripts/includes and not to `source` variables like `WORKER_x_MTR_SUITES`.
I converted `suites-groups.sh` just to consist of 2 functions `get_suites()` and `check_suites()`. The `WORKER_x_MTR_SUITES` variables will be defined only if one of functions was called.

File: jenkins/pipeline-parallel-mtr.groovy (hunk 2 above)
Comment (inikep) [reply jenkins/pipeline-parallel-mtr.groovy]: We source only functions now, not variables. 
https://stackoverflow.com/questions/10822790/can-i-call-a-function-of-a-shell-script-from-another-shell-script

File: jenkins/pipeline-parallel-mtr.groovy (hunk 3 above)
Comment (inikep) [reply jenkins/pipeline-parallel-mtr.groovy:243]: Did you read the topic of this PR?
```
PS-8978: Use different WORKER_x_MTR_SUITES for Debug and RelWithDebInfo
```

Comment (inikep) [reply jenkins/pipeline-parallel-mtr.groovy]: `check_suites` calls `get_suites` to define `WORKER_x_MTR_SUITES` so it needs `${CMAKE_BUILD_TYPE}`,

**Comment by inikep** [comment]
I updated `suites-groups.sh` to use a hybrid approach. The following cases will work identically:
1st use case:
```
suites-groups.sh check ${WORKSPACE}/mysql-test-run.pl ${CMAKE_BUILD_TYPE}
source ${WORKSPACE}/suites-groups.sh
set_suites ${CMAKE_BUILD_TYPE}
```
2nd use case (Kamil's idea with `BUILD_TYPE` bash env variable):
```
export BUILD_TYPE=${CMAKE_BUILD_TYPE}
suites-groups.sh check ${WORKSPACE}/mysql-test-run.pl
source ${WORKSPACE}/suites-groups.sh
```
3rd use case (my preferred approach):
```
source ${WORKSPACE}/suites-groups.sh
set_suites ${CMAKE_BUILD_TYPE}
check_suites ${WORKSPACE}/mysql-test-run.pl
```

**Review by kamil-holubicki — APPROVED** [review]

**Review by percona-ysorokin — APPROVED** [review]
LGTM

**Pull request merged**